addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/MiHomeSubdevicesHandler.java

```diff
@@ -0,0 +1,648 @@
+        }
+
+        // Wait the user to push the button
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_CONFIGURATION_PENDING,
+                "Please push the pairing button on device of type " + deviceType);
+        try {
+            Thread.sleep(PARING_WAIT_TIME_MSEC);
+        } catch (InterruptedException e) {
+            logger.error("Pairing was interrupted ", e);
+            Thread.currentThread().interrupt();
+            return false;
+        }
+
+        logger.debug(
+                "Gathering information about existing devices of type {}, registered on the Mi|Home gateway with id {}",
+                deviceType, gatewayID);
+        // List devices again and see if a new device is added
+        JsonArray subdevicesAfter = gatewayHandler.listSubdevices();
+        if (subdevicesAfter == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    "Unable to get information for subdevices");
+            return false;
+        }
+
+        logger.debug("Searching for new devices of type {}", deviceType);
+        JsonObject newDevice = getNewDevice(subdevicesBefore, subdevicesAfter, deviceType, gatewayID);
+        if (newDevice == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    "Can't find device of type " + deviceType + " for gateway " + gatewayID);
+            return false;
+        }
+
+        finishPairing(newDevice);
+
+        return true;
+    }
+
+    private void finishPairing(JsonObject newDevice) {
+        // Get the device ID and persist it
+        logger.info("Getting the device ID");
+        int deviceID = newDevice.get(DeviceConstants.DEVICE_ID_KEY).getAsInt();
+        Configuration configuration = editConfiguration();
+        this.mihomeID = deviceID;
+        // We persist the device ID, it indicates that the device is already added in the Mi|Home cloud
+        configuration.put(MiHomeBindingConstants.PROPERTY_DEVICE_ID, this.mihomeID);
+        updateConfiguration(configuration);
+
+        updateThingProperties(newDevice);
+
+        // Update the label in the MiHome portal
+        String label = getThing().getLabel();
+        JsonObject updateResponse = gatewayHandler.updateSubdevice(this.mihomeID, label);
+        if (updateResponse == null) {
+            logger.warn("Failed to update label of newly paired subdevice {} to {}", this.mihomeID, label);
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+
+        scheduleRegularUpdate(updateInterval);
+    }
+
+    private void scheduleRegularUpdate(long interval) {
+        if (updateTask != null) {
+            this.updateTask.cancel(true);
+            this.updateTask = null;
+        }
+
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    if (getThing().getStatus() == ThingStatus.ONLINE) {
```

Comment:
I prefer to keep it this way, it something device and API specific. Only one subdevice can't be online in this case.

In the MiHome API the subdevice doesn't have status, we only only the status of the bridge. Once the bridge is online all subdevices become online and vice versa.